### PR TITLE
Combine stats and parse

### PR DESCRIPTION
**Pull request by davlin93**

### CODE REVIEW

**Comment by tonyjhuang** [comment]
update based on code style guide (max line length 80, input/output types for functions) and check with $wapnil about whether or not we can omit creation at date and modification date list. other than that everything else looks good.

In the future we should discuss a debug environment variable or commenting out debug trace messages in prod commits.

File: js/main.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      $('#picture-directory').text(picturePath);
+    });
+    directory.browseForDirectory("Choose the picture directory");
+  });
+
+  /*
+    When submit is clicked on the form, parse all of the input into variables
+    and use the Parse module to parse the ratings file.
+  */
+  $('#input-form').submit(function(e) {
```
Comment (tonyjhuang) [inline js/main.js:35]: I'd actually like to see this function cut down a bit, or have some inline comments.

both for the code style guide and for future readability

File: js/parse.js [hunk 2]
```diff
@@ -3,50 +3,21 @@
...
+  var pictures = [];
 
   /*
     Open the file that was input and parse it into an array of picture objects
   */
-  function parse() {
+  my.getPictures = function(ratingsFile, picturePath) {
     var fileStream = new air.FileStream();
     fileStream.open(ratingsFile, air.FileMode.READ);
     content = String(fileStream.readUTFBytes(fileStream.bytesAvailable));
```
Comment (tonyjhuang) [inline js/parse.js]: var content

File: js/parse.js [hunk 3]
```diff
@@ -67,12 +38,16 @@ $(function() {
       for(i = dataIndex + 1; i < rows.length; i++) {
         var cols = rows[i].split(",");
         path = [picturePath, cols[0]].join("/");
```
Comment (tonyjhuang) [inline js/parse.js]: var path

File: js/parse.js [hunk 4]
```diff
@@ -3,50 +3,21 @@
...
-    e.preventDefault();
-    air.trace("parsing: " + $('#ratings-file').val());
-    ratingsFile = new air.File($('#ratings-file').val());
-    numGroups = $('#num-groups').val();
-    picsPerGroup = $('#pics-per-group').val();
-    avgRating = $('#rating-per-group').val();
-    air.trace("numGroups: " + numGroups + ", picsPerGroup: " + picsPerGroup + ", avgRating: " + avgRating);
-    parse();
-  });
+  var pictures = [];
```
Comment (tonyjhuang) [inline js/parse.js]: Is this module-scoped var necessary? Doesn't seem to be used anywhere else, can be locally-scoped in my.getPictures(...)

**Comment by tonyjhuang** [comment]
made some more comments, would like to see these changes too before merge :>

File: js/main.js (hunk 1 above)
Comment (davlin93) [inline js/main.js:35]: c1707579bf

File: js/parse.js (hunk 2 above)
Comment (davlin93) [inline js/parse.js]: 9103717b8

File: js/parse.js (hunk 3 above)
Comment (davlin93) [inline js/parse.js]: 9103717b8

File: js/parse.js (hunk 4 above)
Comment (davlin93) [inline js/parse.js]: addb673f76

**Comment by davlin93** [comment]
@tonyjhuang updated

**Comment by kehphin** [comment]
+1

**Comment by davlin93** [comment]
@tonyjhuang updated

**Pull request merged**